### PR TITLE
Update frequently-asked-questions.md

### PR DESCRIPTION
**Pull request by BjornW**
I've added a warning about the WordPress REST API being open for all Origins. 
I've paraphrased @rmccue to indicate this was discussed recently and to make clear who said this (instead of me). I've added a link to my plugin as an example for verifying Origins before sending out CORS headers.   

Trac tickets for reference: 
- https://core.trac.wordpress.org/ticket/40011#comment:6
- https://core.trac.wordpress.org/ticket/45477

### CODE REVIEW

**Comment by iandunn** [comment]
It looks like the ticket link is broken: `https://core.trac.wordpress.org/changeset/40600intentional`

**Comment by BjornW** [comment]
@iandunn I fixed the link typo. It should work now.

File: using-the-rest-api/frequently-asked-questions.md [hunk 1]
```diff
@@ -70,3 +70,13 @@ If you're finding that you are sending Authentication headers but the request is
...
 </IfModule>
 ```
+
+## Why is the REST API not verifying the incoming Origin header? 
+The current behaviour of reflecting the incoming Origin header as-is without verification, was an [intentional design decision](https://core.trac.wordpress.org/changeset/40600). This allows **any Origin** to access the REST API resources. WordPress relies on 'nonces' for CSRF protection instead. 
+
+[According to @rmccue on 2018-12-05 paraphrased from Slack](https://wordpress.slack.com/archives/C02RQC26G/p1544026168009900): 
+*"CORS predates the REST API, so yes, it was designed that way intentionally, and nothing big has changed since. Nonces are used for CSRF protection instead. tl;dr: CORS is built for CSRF protection, but WordPress already has a system for that (nonces), so we "disable" CORS as it gets in the way of alternative authentication schemes. The REST API is treated as a public part of your site. It's a design decision to expose data from the REST API to all origins; you should be able to override in plugins easily"*
+
+An example of disabling CORS headers for non-verified Origins can be found here: https://bitbucket.org/BjornW/wp-no-auto-origin-header/src/master/
```
Comment (kadamwhite) [inline using-the-rest-api/frequently-asked-questions.md]: Unfortunately documentation handbook editorial guidelines prevent us from linking to third-party example code like this. I'm going to reword this to explain the general approach you use, and I would encourage you to add your example callback to the [`rest_pre_serve_request` documentation page](https://developer.wordpress.org/reference/hooks/rest_pre_serve_request/) as a user-contributed node.

Comment (BjornW) [reply using-the-rest-api/frequently-asked-questions.md]: Feel free to copy the code verbatim from the external repo, since it's 1) GPL like WordPress and 2) It's my code ;) Thanks.

Comment (kadamwhite) [reply using-the-rest-api/frequently-asked-questions.md]: I don’t believe this page is the place for such a long example, but I’d encourage you again to submit the plugin as a user code example on the filter’s page in the development reference site. Thanks for putting the example together and thank you for contributing to the documentation!
